alternator/executor.cc

```diff
@@ -2989,8 +2989,10 @@ static future<executor::request_return_type> do_query(schema_ptr schema,
 
 static dht::token token_for_segment(int segment, int total_segments) {
     assert(total_segments > 1 && segment >= 0 && segment < total_segments);
-    int64_t delta = std::numeric_limits<int64_t>::max() / total_segments * 2;
-    return dht::token::from_int64(std::numeric_limits<int64_t>::min() + delta * segment);
+    int64_t delta = std::numeric_limits<int64_t>::max() / total_segments;
+    // delta*segment is added twice in order to avoid integer overflow warning.
+    // Generated assembly is identical to numeric_limits<int64_t>::min() + (2 * delta * segment)
+    return dht::token::from_int64(std::numeric_limits<int64_t>::min() + delta * segment + delta * segment);
```

Comment:
while this change is perfectly fine, I wonder, wouldn't everything be easier to understand if you did "delta" in uint64_t instead of int64_t? I.e.,
```c++
uint64_t delta = std::numeric_limits<uint64_t>::max() / total_segments;
return dht::token::from_int64(std::numeric_limits<int64_t>::min() + delta * segment);
```
Any reason why this won't work?
Note no "2" anywhere in my suggestion, I think it's clearer and doesn't need any comments to explain (I think it's even clearer than the old code which had that bizarre "2")

Reply:
I'm pretty sure that it will complain that we try to add something unsigned and bigger than `std::numeric_limits<int64_t>::max()` to `int64`, which can potentially result in overflow (it won't, but not sure if the compiler or ubsan cares enough to figure it out). I haven't verified it though.

Reply:
I don't understand how it can complain this... *any* addition may overflow if the numbers are big enough - it's not only true when adding signed and unsigned numbers...

Reply:
Well, I launched clang++ with ubsan and couldn't make it to complain, so I agree :) I'll send v2.